### PR TITLE
Extract code snippet into stand alone component

### PR DESCRIPTION
**Pull request by charisk**
Extracts components and logic that help us render code snippets out of `AnalysisAlertResult` and into a new stand-alone component so that it can be re-used when showing code paths for problem path queries.

The code is mostly a copy of what has been checked in but with a few adjustments:
- Improved styling 
- Improved highlighting for multi-line highlights
- It no longer receives an alert model
- The "message" part is been made optional, with the ability to receive more children (this is where we'll plug in the code paths link for problem path queries).

Note that `AnalysisAlertResult` is really quite thin now and could be incorporated back into `RemoteQueries.tsx` but I think it'll get a bit more meat to it once we do code paths.

## Checklist
N/A:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+}) => {
+  if (!shouldHighlightLine(lineNumber, highlightedRegion)) {
+    return <PlainLine text={line} />;
+  }
+
+  const isSingleLineHighlight = highlightedRegion.endLine === undefined;
+  const isFirstHighlightedLine = lineNumber === highlightedRegion.startLine;
+  const isLastHighlightedLine = lineNumber === highlightedRegion.endLine;
+
+  const highlightStartColumn = isSingleLineHighlight
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx:133]: Going back to our conversation on the previous PR about handling invalid source locs, there are a number of common errors that I think you can handle right here:

```js
if (highlightStartColumn > highilightEndColumn) {
  ([highlightStartColumn, highilightEndColumn] = [highilightEndColumn, highlightStartColumn]);
}

highlightStartColumn = Math.max(highlightStartColumn, 0);
highilightEndColumn = Math.max(highlightStartColumn, highilightEndColumn);
```

Or some variant of this. It looks like this code won't cause an error if the locations are invalid, but they might cause missing highlights. Again, this is pretty rare, but it does happen.

File: extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    ? highlightedRegion.startColumn
+    : isFirstHighlightedLine
+      ? highlightedRegion.startColumn
+      : 0;
+
+  const highilightEndColumn = isSingleLineHighlight
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx:139]: Typo: `highilightEndColumn` -> `highlightEndColumn`

**Review by aeisenberg — APPROVED** [review]
Approving, assuming you will fix the typo. :)

My comment about the correctness of the locations can be done in the future.

File: extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/FileCodeSnippet.tsx:133]: Thanks for the example/code. Lets discuss validation next week!

**Pull request merged**